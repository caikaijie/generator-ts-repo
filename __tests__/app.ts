'use strict'
import path from 'path'
import helpers from 'yeoman-test'
import dirTree, { DirectoryTree } from 'directory-tree'
import treeify from 'treeify'
import fs from 'fs'

function treeify_(t: DirectoryTree): string {
    const convert = (
        p: treeify.TreeObject,
        node: DirectoryTree
    ): treeify.TreeObject => {
        if (node) {
            let c: treeify.TreeObject = {}
            if (node.children) {
                node.children.forEach(v => {
                    c = convert(c, v)
                })
            }

            p[node.name] = c
            return p
        }

        return {}
    }

    const tj = convert({}, t)

    return treeify.asTree(tj, true, false)
}

describe('generator-ts-repo:app', () => {
    let tmpDir = ''
    const repoName = 'e2e-test0000'

    beforeAll(() => {
        return helpers
            .run(path.join(__dirname, '../generators/app'))
            .withPrompts({ repoName })
            .inTmpDir(dir => {
                tmpDir = dir
            })
    })

    it('creates structure', () => {
        expect(
            treeify_(
                dirTree(path.join(tmpDir, repoName), {
                    normalizePath: true,
                    exclude: /.git\/.*/,
                })
            )
        ).toMatchSnapshot('directory structure')
    })

    it('creates packages.json', () => {
        const content: string = fs
            .readFileSync(path.join(tmpDir, repoName, 'package.json'))
            .toString()
        expect(content).toMatchSnapshot('package.json content')
    })
})
