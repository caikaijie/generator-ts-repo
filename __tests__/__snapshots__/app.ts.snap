// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generator-ts-repo:app creates packages.json: package.json content 1`] = `
"{
  \\"name\\": \\"e2e-test0000\\",
  \\"version\\": \\"0.0.1\\",
  \\"description\\": \\"A git repository with Node and Typescript.\\",
  \\"keywords\\": [
    \\"node\\",
    \\"typescript\\"
  ],
  \\"homepage\\": \\"\\",
  \\"license\\": \\"MIT\\",
  \\"files\\": [
    \\"dist\\"
  ],
  \\"main\\": \\"dist/index.js\\",
  \\"scripts\\": {
    \\"prebuild\\": \\"npm run lint\\",
    \\"build\\": \\"del-cli dist && tsc --outDir dist\\",
    \\"build:debug\\": \\"del-cli dist && tsc --sourceMap --outDir dist\\",
    \\"changelog\\": \\"conventional-changelog -p angular -i CHANGELOG.md -s\\",
    \\"commit\\": \\"git-cz\\",
    \\"lint\\": \\"eslint --fix src/**/*\\",
    \\"test\\": \\"jest\\",
    \\"watch\\": \\"del-cli dist && tsc -w --outDir dist\\"
  },
  \\"husky\\": {
    \\"hooks\\": {
      \\"pre-commit\\": \\"tsc --noEmit && lint-staged\\",
      \\"commit-msg\\": \\"commitlint -E HUSKY_GIT_PARAMS\\"
    }
  },
  \\"lint-staged\\": {
    \\"*.{js,ts,tsx}\\": [
      \\"eslint --fix\\",
      \\"git add\\"
    ]
  },
  \\"config\\": {
    \\"commitizen\\": {
      \\"path\\": \\"./node_modules/cz-conventional-changelog\\"
    }
  },
  \\"dependencies\\": {},
  \\"devDependencies\\": {
    \\"@commitlint/cli\\": \\"^8.2.0\\",
    \\"@commitlint/config-conventional\\": \\"^8.2.0\\",
    \\"@types/jest\\": \\"^24.0.23\\",
    \\"@typescript-eslint/eslint-plugin\\": \\"^2.11.0\\",
    \\"@typescript-eslint/parser\\": \\"^2.11.0\\",
    \\"commitizen\\": \\"^4.0.3\\",
    \\"conventional-changelog-cli\\": \\"^2.0.28\\",
    \\"cz-conventional-changelog\\": \\"3.0.2\\",
    \\"del-cli\\": \\"^3.0.0\\",
    \\"eslint\\": \\"^6.7.2\\",
    \\"eslint-config-prettier\\": \\"^6.7.0\\",
    \\"eslint-plugin-prettier\\": \\"^3.1.1\\",
    \\"husky\\": \\"^3.0.9\\",
    \\"jest\\": \\"^24.8.0\\",
    \\"lint-staged\\": \\"^9.4.3\\",
    \\"prettier\\": \\"^1.19.1\\",
    \\"ts-jest\\": \\"^24.2.0\\",
    \\"typescript\\": \\"^3.7.3\\"
  },
  \\"engines\\": {
    \\"node\\": \\">=8.6.0\\"
  }
}"
`;

exports[`generator-ts-repo:app creates structure: directory structure 1`] = `
"└─ e2e-test0000
   ├─ .editorconfig
   ├─ .eslintignore
   ├─ .eslintrc.js
   ├─ .git
   ├─ .gitattributes
   ├─ .gitignore
   ├─ .prettierrc
   ├─ .travis.yml
   ├─ .vscode
   │  └─ settings.json
   ├─ LICENSE
   ├─ README.md
   ├─ commitlint.config.js
   ├─ jest.config.js
   ├─ package.json
   ├─ src
   │  ├─ example.spec.ts
   │  └─ example.ts
   └─ tsconfig.json
"
`;
